kernel/libs/aster-bigtcp/src/socket/bound/tcp_conn.rs

```diff
@@ -155,25 +247,26 @@ impl<E: Ext> Inner<E> for TcpConnectionInner<E> {
     type Observer = E::TcpEventObserver;
 
     fn on_drop(this: &Arc<SocketBg<Self, E>>) {
```

Comment:
Now both `on_drop` only contains `debug_assert`, maybe it can be removed in this PR or in the future?

Reply:
Well, we need to make sure that the resource doesn't get leaked without notice, right?

Another reason is that UDP sockets are still removed in its `on_drop` method. Dropping UDP sockets does not require `iface.poll()`, so it is simpler and there is no need to create a `close()` method for UDP sockets.